### PR TITLE
[terraform-resources] 5.x provider asg updates (backwards compatible)

### PR DESCRIPTION
**Pull request by maorfr**
a better version of #4196

### CODE REVIEW

File: reconcile/utils/terrascript_aws_client.py [hunk 1]
```diff
@@ -5469,9 +5469,13 @@ def populate_tf_resource_asg(self, spec) -> None:
...
             )
 
-        asg_value["tag"] = [
+        tags = [
             {"key": k, "value": v, "propagate_at_launch": True} for k, v in tags.items()
         ]
+        if self.versions.get(account).startswith("3"):
+            asg_value["tags"] = tags
+        else:
+            asg_value["tag"] = tags
```
Comment (hemslo) [inline reconcile/utils/terrascript_aws_client.py:5478]: I thought both versions support `tag` in the same format, no?

Comment (maorfr) [reply reconcile/utils/terrascript_aws_client.py]: both are supported, but i'm trying to perform a migration that has `No changes. Your infrastructure matches the configuration.` in every step of the way

File: reconcile/utils/terrascript_aws_client.py [hunk 2]
```diff
@@ -5469,9 +5469,13 @@ def populate_tf_resource_asg(self, spec) -> None:
                 override
             )
 
-        asg_value["tag"] = [
+        tags = [
             {"key": k, "value": v, "propagate_at_launch": True} for k, v in tags.items()
         ]
+        if self.versions.get(account, "").startswith("3"):
+            asg_value["tags"] = tags
```
Comment (geoberle) [inline reconcile/utils/terrascript_aws_client.py:5476]: it seems asg 3.x also support `tag` and `tags` but it seems `tags` is something they call interpolated tags 

https://registry.terraform.io/providers/hashicorp/aws/3.73.0/docs/resources/autoscaling_group#interpolated-tags

Comment (geoberle) [reply reconcile/utils/terrascript_aws_client.py:5476]: ah just saw the `No changes` idea. makes sense

**Review by geoberle — APPROVED** [review]

**Review by hemslo — APPROVED** [review]
LGTM!

**Pull request merged**